controllers/spacebindingrequestmigration/spacebindingrequest_migration_controller.go

```diff
@@ -0,0 +1,182 @@
+	for _, memberCluster := range memberClusters {
+		b = b.Watches(
+			source.NewKindWithCache(&toolchainv1alpha1.SpaceBindingRequest{}, memberCluster.Cache),
+			handler.EnqueueRequestsFromMapFunc(MapSpaceBindingRequestToSpaceBinding(r.Client)))
+	}
+	return b.Complete(r)
+}
+
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spacebindingrequests,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spacebindingrequests/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spacebindingrequests/finalizers,verbs=update
+
+// Reconcile converts all the SpaceBindings created using the sandbox-cli to SpaceBindingRequests
+func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("reconciling SpaceBindingRequestMigration")
+
+	// Fetch the SpaceBinding instance
+	spaceBinding := &toolchainv1alpha1.SpaceBinding{}
+	err := r.Client.Get(ctx, request.NamespacedName, spaceBinding)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, errs.Wrapf(err, "unable to get spacebinding")
+	}
+	if util.IsBeingDeleted(spaceBinding) {
+		logger.Info("the SpaceBinding is already being deleted")
+		return reconcile.Result{}, nil
+	}
+	// check if spaceBinding was created from SpaceBindingRequest,
+	// in that case we can skip it
+	if hasSpaceBindingRequest(spaceBinding) {
+		return reconcile.Result{}, nil
+	}
+
+	spaceName := types.NamespacedName{Namespace: spaceBinding.Namespace, Name: spaceBinding.Spec.Space}
+	space := &toolchainv1alpha1.Space{}
+	if err := r.Client.Get(ctx, spaceName, space); err != nil {
+		if errors.IsNotFound(err) {
+			// space was deleted
+			return reconcile.Result{}, nil
+		}
+		// error while reading space
+		return ctrl.Result{}, errs.Wrapf(err, "unable to get the bound Space")
+	}
+
+	murName := types.NamespacedName{Namespace: spaceBinding.Namespace, Name: spaceBinding.Spec.MasterUserRecord}
+	mur := &toolchainv1alpha1.MasterUserRecord{}
+	if err := r.Client.Get(ctx, murName, mur); err != nil {
+		if errors.IsNotFound(err) {
+			// mur was deleted
+			return reconcile.Result{}, nil
+		}
+		// error while reading MUR
+		return ctrl.Result{}, errs.Wrapf(err, "unable to get the bound MUR")
+	}
+
+	// skip workspace creator spacebinding
+	// the controller will convert only spacebindings created by system admins using the sandbox-cli.
+	// If the creator label on the space matches the usersignup reference on the MUR then this is the owner of the space
+	// and the spacebinding should not be migrated.
+	if mur.ObjectMeta.OwnerReferences == nil || len(mur.ObjectMeta.OwnerReferences) == 0 {
```

Comment:
I've tried it [here](https://go.dev/play/p/V23cBGs1SpQ?v=goprev) , but it gives an error 😕

Reply:
that's not working because go can not understand which is the type of the `nil`, hence it's not able to compile.
Try with something similar to [this](https://go.dev/play/p/n2CZGzep_TJ?v=goprev) or [this](https://go.dev/play/p/zlG1KtbummT?v=goprev)

Reply:
hmm good point! Thanks for double checking! I've accepted your suggestions!